core/trace/src/main/scala/org/typelevel/otel4s/trace/Tracer.scala

```diff
@@ -59,6 +59,14 @@ trait Tracer[F[_]] extends TracerMacro[F] {
     */
   def childScope[A](parent: SpanContext)(fa: F[A]): F[A]
 
+  final def childOrContinue[A](parent: Option[SpanContext])(fa: F[A]): F[A] =
```

Comment:
This is more or less Natchez EntryPoint.continueOrElseRoot ... except we already have a default scope.

Reply:
This is a higher-level method that conflates tracing and propagation, but it seems like a common idiom?

```scala
def joinTrace[F[_]: FlatMap: Tracer: TextMapPropagator: Local[*[_], Vault], A: TextMapGetter, B](carrier: A)(fb: F[B]): F[B] =
    Local[F, Vault].ask[Vault].flatMap { vault =>
      SpanContext.fromContext(TextMapPropagator[F].extract(vault, carrier)) match {
        case Some(parent) =>
          Tracer[F].childScope(parent)(fb)
        case None =>
          fb
      }
    }
```

Reply:
Technically, we can offer a utility method

```scala
trait Tracer[F[_]] {
  def joinOrContinue[A: TextMapGetter](carrier: A)(fa: F[A]): F[A]
}
```